src/software/new_geom/util/distance.h

```diff
@@ -1,14 +1,76 @@
 #pragma once
 
+#include <limits>
+
+#include "software/geom/util.h"
 #include "software/new_geom/line.h"
 #include "software/new_geom/point.h"
+#include "software/new_geom/polygon.h"
+#include "software/new_geom/segment.h"
 
 /**
  * Finds the shortest distance between a Line and a Point
  *
- * @param first the Line
- * @param second the Point
+ * @param first
+ * @param second
  * @return the shortest distance between first and second
  */
 double distance(const Line &first, const Point &second);
 double distance(const Point &first, const Line &second);
+
+/**
+ * Finds the shortest distance between two Points
+ *
+ * @param first
+ * @param second
+ * @return the shortest distance between first and second
+ */
+double distance(const Point &first, const Point &second);
+
+/**
+ * Finds the shortest distance between two Segments
+ *
+ * @param first
+ * @param second
+ * @return the shortest distance between first and second
+ */
+double distance(const Segment &first, const Segment &second);
+
+/**
+ * Finds the shortest distance between a Point and a Segment
+ *
+ * @param first
+ * @param second
+ * @return the shortest distance between first and second
+ */
+double distance(const Point &first, const Segment &second);
+double distance(const Segment &first, const Point &second);
+
+/**
+ * Finds the shortest distance between a Point and a Polygon
```

Comment:
nit. Maybe add that points inside the polygon are considered to have a distance of 0? I can just see that being a point of confusion for people reading this for the first time.

Reply:
Done